packages/@ama-sdk/core/src/fwk/param-serialization.ts

```diff
@@ -246,3 +289,172 @@ export function serializePathParams<T extends { [key: string]: SupportedParamTyp
+      deserializedValue = serializedParamValue.split('=')[1];
+    }
+    if (deserializedValue) {
+      acc[queryParamName as keyof T] = deserializedValue;
+    }
+    return acc;
+  }, {} as { [p in keyof T]: SupportedParamType });
+}
+
+/**
+ * Deserialize path parameters of type primitive
+ * OpenAPI Parameter Serialization {@link https://swagger.io/specification | documentation}
+ * @param serializedParamValue serialized path parameter value
+ * @param paramSerialization parameter serialization
+ */
+function deserializePrimitivePathParams(serializedParamValue: string, paramSerialization: ParamSerialization) {
+  switch (paramSerialization.style) {
+    case 'simple': {
+      return decodeURIComponent(serializedParamValue);
+    }
+    case 'label': {
+      return decodeURIComponent(serializedParamValue.substring(1));
+    }
+    case 'matrix': {
+      return decodeURIComponent(serializedParamValue.substring(1).split('=')[1]);
+    }
+  }
+}
+
+/**
+ * Deserialize path parameters of type array
+ * OpenAPI Parameter Serialization {@link https://swagger.io/specification | documentation}
+ * @param serializedParamValue serialized path parameter value
+ * @param paramSerialization parameter serialization
+ */
+export function deserializeArrayPathParams(serializedParamValue: string, paramSerialization: ParamSerialization) {
+  switch (paramSerialization.style) {
+    case 'simple': {
+      return serializedParamValue.split(',');
+    }
+    case 'label': {
+      return paramSerialization.explode ? serializedParamValue.substring(1).split('.') : serializedParamValue.substring(1).split(',');
+    }
+    case 'matrix': {
+      return paramSerialization.explode
+        ? serializedParamValue.substring(1).split(';').map((value) => value.split('=')[1])
+        : serializedParamValue.split('=')[1].split(',');
+    }
+  }
+}
+
+/**
+ * Deserialize path parameters of type object
+ * OpenAPI Parameter Serialization {@link https://swagger.io/specification | documentation}
+ * @param serializedParamValue serialized path parameter value
+ * @param paramSerialization parameter serialization
+ */
+export function deserializeObjectPathParams(serializedParamValue: string, paramSerialization: ParamSerialization) {
+  const splitObject: string[] = paramSerialization.style === 'matrix' && paramSerialization.explode
+    ? serializedParamValue.substring(1).split(';')
+    : deserializeArrayPathParams(serializedParamValue, paramSerialization)!;
+  if (paramSerialization.explode) {
+    return splitObject.reduce((obj: { [key: string]: PrimitiveType }, currentValue) => {
+      const [key, value] = currentValue.split('=');
+      obj[key] = value;
+      return obj;
+    }, {} as { [key: string]: PrimitiveType });
+  }
+  return splitObject.reduce((obj: { [key: string]: PrimitiveType }, currentValue, index, array) => {
+    if (index % 2 === 0) {
+      obj[currentValue] = array[index + 1];
+    }
+    return obj;
+  }, {} as { [key: string]: PrimitiveType });
```

Comment:
since this is used several times we could extract the code in a dedicated function
```typescript
export function objectFromPairwiseArray<T>(inputArray: T[]): Record<string, T> {
  return inputArray.reduce((obj: { [key: string]: T}, currentValue, index, array) => {
    if (index % 2 === 0) {
      obj[currentValue] = array[index + 1];
    }
    return obj;
  }, []);
}
```
```suggestion
  return objectFromPairwiseArray(splitObject);
```
or something like that

Reply:
this is used only twice (deserialization of object query and path parameters), I am not sure if it is worth creating another function in the code but if it makes it easier to read/understand then yes I can make the update